lib/configurationSchema.js

```diff
@@ -0,0 +1,438 @@
+        },
+        {
+          type: 'string',
+        },
+      ],
+    },
+    importStatementFormatter: {
+      instanceof: 'Function',
+      default: ({ importStatement }) => importStatement,
+    },
+    logLevel: {
+      default: 'info',
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          default: 'info',
+          type: 'string',
+          enum: ['debug', 'info', 'warn', 'error'],
+        },
+      ],
+    },
+    maxLineLength: {
+      default: 80,
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'integer',
+          minimum: 10,
+        },
+      ],
+    },
+    mergableOptions: {
+      type: 'object',
+      default: {
+        aliases: true,
+        coreModules: true,
+        namedExports: true,
+        globals: true,
+      },
+      additionalProperties: false,
+      properties: {
+        aliases: { type: 'boolean' },
+        coreModules: { type: 'boolean' },
+        danglingCommas: { type: 'boolean' },
+        declarationKeyword: { type: 'boolean' },
+        environments: { type: 'boolean' },
+        excludes: { type: 'boolean' },
+        globals: { type: 'boolean' },
+        groupImports: { type: 'boolean' },
+        ignorePackagePrefixes: { type: 'boolean' },
+        importDevDependencies: { type: 'boolean' },
+        importFunction: { type: 'boolean' },
+        importStatementFormatter: { type: 'boolean' },
+        logLevel: { type: 'boolean' },
+        maxLineLength: { type: 'boolean' },
+        minimumVersion: { type: 'boolean' },
+        moduleNameFormatter: { type: 'boolean' },
+        namedExports: { type: 'boolean' },
+        sortImports: { type: 'boolean' },
+        stripFileExtensions: { type: 'boolean' },
+        tab: { type: 'boolean' },
+        useRelativePaths: { type: 'boolean' },
+      },
+    },
+    minimumVersion: {
+      default: '0.0.0',
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'string',
+        },
+      ],
+    },
+    moduleNameFormatter: {
+      instanceof: 'Function',
+      default: ({ moduleName }) => moduleName,
+    },
+    moduleSideEffectImports: {
+      instanceof: 'Function',
+      default: () => [],
+    },
+    namedExports: {
+      default: {},
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'object',
+          additionalProperties: {
+            type: 'array',
+            items: {
+              type: 'string',
+            },
+          },
+        },
+      ],
+    },
+    packageDependencies: {
+      default: ({ config }) =>
+        findPackageDependencies(
+          config.workingDirectory,
+          config.get('importDevDependencies')
+        ),
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'array',
+          items: {
+            type: `string`,
```

Comment:
Should probably change the back ticks to single quotes

Reply:
Ah, old habits die hard.

In a previous company, we resolved the "single quotes vs double quotes" debate with "always use backticks".

I've changed it to single quotes now.